### PR TITLE
Mask secret value rendered as planpreview results

### PR DESCRIPTION
**Pull request by khanhtc1202**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
Mask secret and configuration values rendered as plan-preview result
```

### CODE REVIEW

File: pkg/app/piped/cloudprovider/kubernetes/diff.go [hunk 1]
```diff
@@ -118,15 +118,19 @@ func (r *DiffListResult) Render(opt DiffRenderOptions) string {
 		index++
 		b.WriteString(fmt.Sprintf("# %d. %s\n\n", index, key.ReadableString()))
 
-		if opt.UseDiffCommand {
+		// Use our diff check in the following cases:
+		// - not explicit set useDiffCommand option.
+		// - requires masking secret or configmap value.
+		if !opt.UseDiffCommand || opt.MaskSecret || opt.MaskConfigMap {
```
Comment (nghialv) [inline pkg/app/piped/cloudprovider/kubernetes/diff.go:124]: I think we have to use `diff` command even if `opt.MaskSecret` is true and the resource is not a Secret.

Comment (khanhtc1202) [reply pkg/app/piped/cloudprovider/kubernetes/diff.go:124]: Ah, I get your point, will update the condition

Comment (nghialv) [reply pkg/app/piped/cloudprovider/kubernetes/diff.go:124]: So basically we need a flag configured at lines 112 and 114.

Comment (nghialv) [reply pkg/app/piped/cloudprovider/kubernetes/diff.go:124]: Btw, could you add an issue about adding more tests around this part?

Comment (khanhtc1202) [reply pkg/app/piped/cloudprovider/kubernetes/diff.go]: PTAL https://github.com/pipe-cd/pipe/pull/2423/commits/c76d206ba37ac29efd13ae6280c8b39bac3bfc8b

**Comment by nghialv** [comment]
Thank you.
/lgtm

**Pull request merged**